### PR TITLE
Zen 132 update tap resolver

### PR DESCRIPTION
**Pull request by lancetipton**
**Submitted** - 2/10/2020

### Important
  * This pull request is should reviewed along side the following other pull requests
    * [tap-events-force](https://github.com/simpleviewinc/tap-events-force/pull/4)
    * [sv-keg](https://github.com/simpleviewinc/sv-keg-expo/pull/6)
    * [sv-tap-resolver](https://github.com/simpleviewinc/sv-tap-resolver/pull/10)

### Changes
  * This is basically a complete rewrite
  * There are so many changes, it might as well be a separate repo
  * I'm not sure it's worth doing a compare with old code, but feel free to review the code
  * I added a lot of tests, but it still no fully complete
  * There's so many changes that I want to get it merged in, and can come back to it
  * We can ensure it working properly by doing the tests below for both web and mobile
  * Once it's merged, we can start versioning
    * This will allow for better changes and pull requests moving forward
  * It also adds a demo app to show the different features
    * I'll deploy once this is merged

### Test
  * The Best way to test all of these changes to **All Pull Requests** listed above
    * Pull this repo down into a new folder
      * `git clone https://github.com/lancetipton/tap-events-force.git test-the-tap`
    * Navigate to the new directory
      * `cd test-the-tap`
    * Setup the repos
      * `yarn setup`
    * Run the web version and the ios version
      * `yarn web`
      * `yarn ios`

### CODE REVIEW

File: app/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Re-Theme Example
```
Comment (daniel112) [inline app/README.md]: the example usage seems to be slightly different than how it's used now. maybe we should update the README

<ReThemeProvider value={ this.state.theme } > vs ReThemeProvider theme={{}} merge={true}

Comment (mpcarolin) [reply app/README.md]: > I pulled this repo > switched branch > `yarn install` > `yarn test`
> 
> it failed 1 test out of 72 on `src/context/__tests__/provider.js` > `should set the state value to be the current dimensions`

The same test failed for me.

File: src/context/__tests__/provider.js [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+    expect(stateValue).toBe(null)
+    renderReThemeProvider()
+
+    expect(typeof stateValue).toBe('object')
+
+    Object.keys(dimensionData.window)
+      .map(key => expect(stateValue[key]).not.toBe(undefined))
+
+  })
```
Comment (daniel112) [inline src/context/__tests__/provider.js]: test failed here for me

File: app/README.md (hunk 1 above)
Comment (lancetipton) [reply app/README.md]: The test has been fixed

File: src/context/__tests__/provider.js (hunk 2 above)
Comment (lancetipton) [reply src/context/__tests__/provider.js]: This is fixed

Comment (lancetipton) [reply src/context/__tests__/provider.js]: README.md has also been updated

**Pull request merged**